tests/blackbox/testlib/dbus.py

```diff
@@ -0,0 +1,232 @@
+    _POOL_IFACE = "org.storage.stratis1.pool"
+    _FS_IFACE = "org.storage.stratis1.filesystem"
+
+    _DBUS_TIMEOUT_SECONDS = 120
+    _TIMEOUT = _get_timeout(
+        os.environ.get("STRATIS_DBUS_TIMEOUT", _DBUS_TIMEOUT_SECONDS * 1000)
+    )
+
+    @staticmethod
+    def _get_managed_objects():
+        """
+        Get managed objects for stratis
+        :return: A dict,  Keys are object paths with dicts containing interface
+                          names mapped to property dicts.
+                          Property dicts map names to values.
+        """
+        object_manager = dbus.Interface(
+            StratisDbus._BUS.get_object(StratisDbus._BUS_NAME, StratisDbus._TOP_OBJECT),
+            StratisDbus._OBJECT_MANAGER,
+        )
+        return object_manager.GetManagedObjects(timeout=StratisDbus._TIMEOUT)
+
+    @staticmethod
+    def pool_list():
+        """
+        Query the pools
+        :return: A list of pool names.
+        """
+        pool_objects = [
+            obj_data[StratisDbus._POOL_IFACE]
+            for _, obj_data in StratisDbus._get_managed_objects().items()
+            if StratisDbus._POOL_IFACE in obj_data
+            and obj_data[StratisDbus._POOL_IFACE]["Name"].startswith(TEST_PREF)
+        ]
+
+        return [pool_obj["Name"] for pool_obj in pool_objects]
+
+    @staticmethod
+    def pool_destroy(pool_name):
+        """
+        Destroy a pool
+        :param name: Name of pool to destroy
+        :return: None
+        """
+        pool_objects = {
+            path: obj_data[StratisDbus._POOL_IFACE]
+            for path, obj_data in StratisDbus._get_managed_objects().items()
+            if StratisDbus._POOL_IFACE in obj_data
+        }
+
+        pool_object_paths = [
+            path
+            for path, pool_obj in pool_objects.items()
+            if pool_obj["Name"] == pool_name and pool_obj["Name"].startswith(TEST_PREF)
+        ]
+        if pool_object_paths == []:
+            return None
+
+        iface = dbus.Interface(
+            StratisDbus._BUS.get_object(StratisDbus._BUS_NAME, StratisDbus._TOP_OBJECT),
+            StratisDbus._MNGR_IFACE,
+        )
+        iface.DestroyPool(pool_object_paths[0], timeout=StratisDbus._TIMEOUT)
+
+        return None
+
+    @staticmethod
+    def fs_list():
+        """
+        Query the file systems
+        :return: A dict,  Key being the fs name, the value being the pool name
+        """
+        objects = StratisDbus._get_managed_objects().items()
+
+        fs_objects = [
+            obj_data[StratisDbus._FS_IFACE]
+            for _, obj_data in objects
+            if StratisDbus._FS_IFACE in obj_data
+            and obj_data[StratisDbus._FS_IFACE]["Name"].startswith(TEST_PREF)
+        ]
+
+        pool_path_to_name = {
+            obj: obj_data[StratisDbus._POOL_IFACE]["Name"]
+            for obj, obj_data in objects
+            if StratisDbus._POOL_IFACE in obj_data
+        }
+
+        result = {
```

Comment:
Please just return the value here, as ```return {...}```. I like to avoid coming up with identifier names for expressions that are used only once, as I think they clutter up the code.

Reply:
Resolved.